### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by dfed**
Exercises more of the SafeDI library via tests, and discovers/fixes an issue where an source empty file list in the CSV would cause an error.

There's also a little code cleanup along the way here.

### CODE REVIEW

File: Sources/SafeDICore/Visitors/InstantiableVisitor.swift [hunk 1]
```diff
@@ -57,9 +57,8 @@ public final class InstantiableVisitor: SyntaxVisitor {
             let patterns = node
                 .bindings
                 .filter {
-                    $0.initializer == nil
```
Comment (dfed) [inline Sources/SafeDICore/Visitors/InstantiableVisitor.swift:60]: this check was duplicative with `isOptionalOrInitialized` below

File: Sources/SafeDICore/Visitors/InstantiableVisitor.swift [hunk 2]
```diff
@@ -57,9 +57,8 @@ public final class InstantiableVisitor: SyntaxVisitor {
...
-                    && $0.accessorBlock == nil
-                    && !$0.isOptionalAndUninitialized
```
Comment (dfed) [inline Sources/SafeDICore/Visitors/InstantiableVisitor.swift:62]: `isOptionalAndUninitialized` was named incorrectly. It was really checking whether something was optional _or_ uninitialized

File: Tests/SafeDICoreTests/TypeDescriptionTests.swift [hunk 3]
```diff
@@ -542,9 +542,9 @@ final class TypeDescriptionTests: XCTestCase {
         XCTAssertEqual(typeDescription.asSource, "() async throws -> ()")
     }
 
-    func test_asSource_whenDescribingAnUnknownCase_returnsTheProvidedStringWithWhitespaceStripped() {
```
Comment (dfed) [inline Tests/SafeDICoreTests/TypeDescriptionTests.swift:545]: turns out we're not stripping leading whitespace. But that's honestly fine since we have no valid, compiling cases that result in an `.unknown` type.

This test wasn't previously testing the `.unknown` case, but rather the `.simple` case.

**Comment by dfed** [comment]
Merging this without review since we aren't changing API – we're just improving coverage.
